plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/ui/AccountSelector.java

```diff
@@ -93,22 +82,36 @@ public boolean isEmailAvailable(String email) {
    * (By its contract, {@link Account} never carries a {@code null} {@link Credential}.)
    */
   public Credential getSelectedCredential() {
-    return selectedAccount != null ? selectedAccount.getOAuth2Credential() : null;
+    Optional<Account> account = getSelectedAccount();
+    return account.isPresent() ? account.get().getOAuth2Credential() : null;
   }
 
   /**
    * Returns the currently selected email, or empty string if none; never {@code null}.
    */
   public String getSelectedEmail() {
-    try {
-      return apiFactory.getAccount().getEmail();
-    } catch (IOException ex) {
-      return "";
+    Optional<Account> account = getSelectedAccount();
+    return account.isPresent() ? account.get().getEmail() : "";
+  }
+  
+  private Optional<Account> getSelectedAccount() {
```

Comment:
Added logic to `fireSelectionListeners()`. If the user logs in with a different account in gcloud, then this will trigger an account "selection", saving us from refactoring the account change logic in dependent components.

Reply:
This function is also the central point that fires the events. All other methods in this class, if they ever require account info, will rely on this method to ensure the event is fired